server/etcdserver/txn/txn.go

```diff
@@ -256,11 +256,15 @@ func Txn(ctx context.Context, lg *zap.Logger, rt *pb.TxnRequest, txnModeWriteWit
 		},
 		"compare",
 	)
-	err := checkTxn(ctx, txnRead, rt, isWrite, lessor, txnPath)
+	if isWrite {
+		trace.AddField(traceutil.Field{Key: "read_only", Value: false})
+	}
+	_, err := checkTxn(txnRead, rt, lessor, txnPath)
 	if err != nil {
 		txnRead.End()
 		return nil, nil, err
 	}
+	trace.Step("check requests")
```

Comment:
Minor comment: put the trace into the function `checkTxn`. All the existing trace items are bound with the operations/functions.

Reply:
Yes, but `checkTxn` is recursive, which means it would call `trace.Step("check requests")` for each subtransaction which is not what we want.